src/app/select-session/page.tsx

```diff
@@ -0,0 +1,228 @@
+/**
+ * @author David Zhu <dzhu292@aucklanduni.ac.nz>
+ */
+
+"use client";
+
+import Heading from "@/components/Heading/Heading";
+import Button from "@/components/Button/Button";
+import { useEffect, useState, useRef, ChangeEvent } from "react";
+import { CgProfile } from "react-icons/cg";
+import { AiFillCaretDown } from "react-icons/ai";
+
+import SessionCard from "@/components/SessionCard/SessionCard";
+import { SessionCardStatus } from "@/components/SessionCard/SessionCardStatusEnum";
+import SessionCardProps from "@/components/SessionCard/SessionCardProps";
+import ScrollShadow from "@/components/ScrollShadow";
+import { twJoin } from "tailwind-merge";
+import { useQuery } from "@tanstack/react-query";
+import type { GameSession } from "@prisma/client";
+
+export default function SelectSessionPage() {
+  const remainingSessions = 11;
+  const isMember = true;
+  const firstName = "David";
+
+  const { data } = useQuery({
+    queryKey: ['current-sessions'],
+    queryFn: async () => {
+      const response = await fetch("api/gamesession/current", {
+        cache: "no-store",
+      });
+      return await response.json();
```

Comment:
```suggestion
      return await response.json() as SessionCardProps[] | undefined;
```

Then you don't have to do the crazy typing underneath. Try to infer your types where possible or else your type defs like underneath might be incorrect.

Could use zod to safeParse (from `/lib/validators.ts`) this object so we always get the correct object shape